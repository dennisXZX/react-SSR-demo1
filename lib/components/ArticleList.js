import React, { Fragment } from 'react'
import Article from './Article';

const ArticleList = (props) => {
  return (
    <Fragment>
      {Object.values(props.articles).map((article) => (
        <Article
          key={article.id}
          article={article}
        />
      ))}
    </Fragment>
  )
}

export default ArticleList
