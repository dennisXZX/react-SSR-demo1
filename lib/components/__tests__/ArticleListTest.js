import React from 'react';
import ArticleList from '../ArticleList';
import Article from '../Article';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ArticleList', () => {
  let wrapper;

  // shallow render the component before testing
  beforeEach(() => {
    wrapper = shallow(<ArticleList {...testProps} />);
  });

  // prepare the props for the component
  const testProps = {
    articles: {
      articleA: { id: 'articleA' },
      articleB: { id: 'articleB' }
    }
  }

  Article.propTypes = {};

  it('should render correctly', () => {
    expect(wrapper.find(Article)).toHaveLength(2);

    // create a snapshot
    expect(wrapper).toMatchSnapshot();
  })
})