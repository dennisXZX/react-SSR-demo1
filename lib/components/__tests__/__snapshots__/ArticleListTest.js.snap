// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "articleA": Object {
          "id": "articleA",
        },
        "articleB": Object {
          "id": "articleB",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Article
          article={
            Object {
              "id": "articleA",
            }
          }
        />,
        <Article
          article={
            Object {
              "id": "articleB",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "articleA",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "id": "articleA",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "articleB",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "id": "articleB",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Article
            article={
              Object {
                "id": "articleA",
              }
            }
          />,
          <Article
            article={
              Object {
                "id": "articleB",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "articleA",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "id": "articleA",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "articleB",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "id": "articleB",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
